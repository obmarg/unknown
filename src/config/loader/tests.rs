use miette::{GraphicalReportHandler, GraphicalTheme};

use super::*;

#[test]
fn test_load_config_from_cwd() {
    insta::assert_debug_snapshot!(load_config_from_path("sample-monorepo/".into()));
}

#[test]
fn test_malformed_project_file() {
    let err =
        load_config_from_path("src/config/loader/test-data/malformed-project/".into()).unwrap_err();

    let mut report = String::new();
    GraphicalReportHandler::new_themed(GraphicalTheme::unicode_nocolor())
        .with_width(80)
        .render_report(&mut report, err.as_ref())
        .unwrap();

    insta::assert_display_snapshot!(report);
}

#[test]
fn test_missing_task_file_error() {
    let err =
        load_config_from_path("src/config/loader/test-data/missing-task-file".into()).unwrap_err();

    let mut report = String::new();
    GraphicalReportHandler::new_themed(GraphicalTheme::unicode_nocolor())
        .with_width(80)
        .render_report(&mut report, err.as_ref())
        .unwrap();

    insta::assert_display_snapshot!(report);
}
