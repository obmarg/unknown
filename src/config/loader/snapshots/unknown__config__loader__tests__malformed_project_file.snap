---
source: src/config/loader/tests.rs
assertion_line: 21
expression: report
---

  × Error parsing project/project.kdl

Error: 
  × child node `project` is required

